### PR TITLE
React api

### PR DESCRIPTION
**Pull request by pabloccid**
# Summary

- I added API requests

# Trello Card
https://trello.com/c/4GU0fVDr

### CODE REVIEW

File: react/src/components/BookList/index.js [hunk 1]
```diff
@@ -2,29 +2,41 @@ import React, { Component } from 'react';
...
-              book[filterField].toLowerCase().includes(filterText.toLowerCase())
-            ){
-              return <BookItem key={book.id} {...book} />;
-            }
-            return null;
-          })
-        }
-      </div>
-    );
+    if(this.state.data){
```
Comment (mvbattan) [inline react/src/components/BookList/index.js]: I'd initialize this as `[]` so we render the list directly

File: react/src/components/BookList/index.js [hunk 2]
```diff
@@ -2,29 +2,41 @@ import React, { Component } from 'react';
...
 import {data} from '../../mock.js';
+import booksGet from '../../services/booksGet';
 
 import './styles.css';
 
 class BookList extends Component {
+  state = {data: null};
+  componentWillMount(){
+    booksGet(
+      sessionStorage.getItem('access_token')
```
Comment (mvbattan) [inline react/src/components/BookList/index.js]: Once the login is successful, I'd set the API header with this value so we avoid putting it every time we fetch data from the backend

File: react/src/screens/BookDetail/index.js [hunk 3]
```diff
@@ -6,25 +6,38 @@ import BookDetailData from '../../components/BookDetailData';
...
-        <div className="book-detail">
-          <BookDetailData bookId={this.props.match.params.id} />
-          <BookDetailSuggestions />
-          <BookDetailCommentaries  bookId={this.props.match.params.id}/>
-        </div>
-      </Fragment>
+  state = ({data: ''})
+  componentWillMount() {
+    bookGet(
+      sessionStorage.getItem('access_token'),
```
Comment (mvbattan) [inline react/src/screens/BookDetail/index.js]: Same here

File: react/src/services/bookGet.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import api from './config';
+
+const bookGet = (Authorization, bookId) => {
+  api.setHeader('Authorization', Authorization);
```
Comment (mvbattan) [inline react/src/services/bookGet.js]: This shouldn't be needed to be performed on every call

File: react/src/services/booksGet.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import api from './config';
```
Comment (mvbattan) [inline react/src/services/booksGet.js]: In general, we put a service per group of calls. These group are defined by the business logic, in this case, `books`. `/books` and `/books/id` could be both in the same file.
Same with `login`, `logout`, `signup`, these are referred as `auth`

File: react/src/services/config.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import {create} from 'apisauce'
```
Comment (mvbattan) [inline react/src/services/config.js]: Also, we usually have a folder named `config` where we drop files like this :)

**Review by mvbattan — CHANGES_REQUESTED** [review]
Just a few comments.
I'm not requesting changes on api calls inside components since we have to refactor them while starting with async actions in redux :)

File: react/src/App.js [hunk 7]
```diff
@@ -8,13 +8,16 @@ import Login from './screens/Login';
...
+import api from './config/service';
 
 import routes from './constants/routes';
 import './styles.css';
 
 class App extends Component {
 
-
+  componentWillMount(){
+    api.setHeader('Authorization', sessionStorage.getItem('Authorization'));
```
Comment (mvbattan) [inline react/src/App.js:19]: Just checking, what if `sessionStorage` doesn't have an `Authorization` key ?

File: react/src/components/BookList/index.js (hunk 1 above)
Comment (mvbattan) [reply react/src/components/BookList/index.js]: Ping here

File: react/src/screens/BookDetail/index.js [hunk 8]
```diff
@@ -6,19 +6,28 @@ import BookDetailData from '../../components/BookDetailData';
...
 import './styles.css';
 import strings from './strings';
 
 class BookDetail extends Component {
+  state = ({data: []})
+  componentWillMount() {
+    getBook(
+      this.props.match.params.id
+    ).then(
+      response => this.setState({data: response.data})
```
Comment (mvbattan) [inline react/src/screens/BookDetail/index.js]: `response.ok &&` ?

**Review by mvbattan — APPROVED** [review]
As a next step, we could manage states in the store

File: react/src/App.js (hunk 7 above)
Comment (pabloccid) [reply react/src/App.js:19]: Then null is assigned to the Authorization header.

**Pull request merged**